homeassistant/components/mqtt/__init__.py

```diff
@@ -868,3 +899,32 @@ def discovery_callback(payload):
                 self.hass,
                 MQTT_DISCOVERY_UPDATED.format(self._discovery_hash),
                 discovery_callback)
+
+
+class MqttEntityDeviceInfo(Entity):
+    """Mixin used for mqtt platforms that support the device registry."""
+
+    def __init__(self, device_config: Optional[ConfigType]) -> None:
+        """Initialize the device mixin."""
+        self._device_config = device_config
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        if not self._device_config:
+            return None
+
+        return {
+            'identifiers': {
+                (DOMAIN, id_)
```

Comment:
All identifiers will have the domain `mqtt`. I don't fully understand what the first index of the tuple means. Can there be multiple identifiers with the same "key"? For example like this:

```python
# output of device_info - should this be valid?
return {'identifiers': {('mqtt', 'identifierA'), ('mqtt', 'identifierB')}}
```

Reply:
yeah can be multiple with same key. I think? 😛 

For example some devices get identifiers in API (Z-Wave node id) but we also have access to a serial number.